src/gui/widgets/Fader.cpp

```diff
@@ -315,154 +284,207 @@ void Fader::updateTextFloat()
 }
 
 
-inline int Fader::calculateDisplayPeak( float fPeak )
-{
-	int peak = static_cast<int>(m_back.height() - (fPeak / (m_fMaxPeak - m_fMinPeak)) * m_back.height());
-
-	return qMin(peak, m_back.height());
-}
-
-
 void Fader::paintEvent( QPaintEvent * ev)
 {
 	QPainter painter(this);
 
-	// Draw the background
-	painter.drawPixmap(ev->rect(), m_back, ev->rect());
-
 	// Draw the levels with peaks
-	if (getLevelsDisplayedInDBFS())
-	{
-		paintDBFSLevels(ev, painter);
-	}
-	else
-	{
-		paintLinearLevels(ev, painter);
-	}
+	// TODO LMMS Compressor, EQ and Delay still use linear displays...
+	paintLevels(ev, painter, !getLevelsDisplayedInDBFS());
 
 	// Draw the knob
-	painter.drawPixmap(0, knobPosY() - m_knob.height(), m_knob);
+	painter.drawPixmap((width() - m_knob.width()) / 2, knobPosY() - m_knob.height(), m_knob);
 }
 
-void Fader::paintDBFSLevels(QPaintEvent * ev, QPainter & painter)
+void Fader::paintLevels(QPaintEvent * ev, QPainter & painter, bool linear)
 {
-	int height = m_back.height();
-	int width = m_back.width() / 2;
-	int center = m_back.width() - width;
+	std::function<float(float value)> mapper = [this](float value) { return ampToDbfs(qMax<float>(0.0001, value)); };
```

Comment:
Why not `auto mapper = ...`?
You can see I did this in the [Piano Roll](https://github.com/LMMS/lmms/blob/master/src/gui/editors/PianoRoll.cpp#L3111).

Reply:
Gotcha, then the only thing I have to say is you don't need `value` in `std::function<float(float value)>`. It should just be `std::function<float(float)>`